[Feature]: Remove peer dependency to `@angular/platform-browser-dynamic`

### 🚀 Feature Proposal

Modern Angular apps don't use `@angular/platform-browser-dynamic` anymore, but only `@angular/platform-browser`.
https://github.com/thymikee/jest-preset-angular/pull/3247 replaced some code in `jest-preset-angular`, but kept the peer dependency in: https://github.com/thymikee/jest-preset-angular/blob/f639f913b1731be8c17a9152944c72f3d9364439/package.json#L65

In case it's still required for some use cases, maybe it's an option to mark the peer dependency as optional?

### Motivation


I would like to remove `@angular/platform-browser-dynamic` from my app's dependencies, but then it often runs into version conflicts during (Renovate) dependency updates, because `jest-preset-angular` still requires `@angular/platform-browser-dynamic` as peer dependency which isn't updated alongside the direct dependencies.

### Example

```txt
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: @angular/platform-browser-dynamic@20.3.14
npm error Found: @angular/common@20.3.15
npm error node_modules/@angular/common
npm error   @angular/common@"20.3.15" from the root project
npm error   peer @angular/common@"20.3.15" from @angular/platform-browser@20.3.15
npm error   node_modules/@angular/platform-browser
npm error     @angular/platform-browser@"20.3.15" from the root project
npm error     peerOptional @angular/platform-browser@"^20.0.0" from @angular/build@20.3.12
npm error     node_modules/@angular/build
npm error       dev @angular/build@"20.3.12" from the root project
npm error     2 more (@angular/router, jest-preset-angular)
npm error   2 more (@angular/router, @ngx-translate/core)
npm error
npm error Could not resolve dependency:
npm error peer @angular/common@"20.3.14" from @angular/platform-browser-dynamic@20.3.14
npm error node_modules/@angular/platform-browser-dynamic
npm error   peer @angular/platform-browser-dynamic@">=18.0.0 <21.0.0" from jest-preset-angular@15.0.3
npm error   node_modules/jest-preset-angular
npm error     dev jest-preset-angular@"15.0.3" from the root project
npm error
npm error Conflicting peer dependency: @angular/common@20.3.14
npm error node_modules/@angular/common
npm error   peer @angular/common@"20.3.14" from @angular/platform-browser-dynamic@20.3.14
npm error   node_modules/@angular/platform-browser-dynamic
npm error     peer @angular/platform-browser-dynamic@">=18.0.0 <21.0.0" from jest-preset-angular@15.0.3
npm error     node_modules/jest-preset-angular
npm error       dev jest-preset-angular@"15.0.3" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```

### Related

- #3247
- #3242
- #3239

--- a/examples/example-app-monorepo/package.json
+++ b/examples/example-app-monorepo/package.json
@@ -21,7 +21,6 @@
     "@angular/core": "^21.0.7",
     "@angular/forms": "^21.0.2",
     "@angular/platform-browser": "^21.0.2",
-    "@angular/platform-browser-dynamic": "^21.0.2",
     "@angular/router": "^21.0.2",
     "angular-in-memory-web-api": "^0.21.0",
     "rxjs": "^7.8.2",

--- a/examples/example-app-v20/package.json
+++ b/examples/example-app-v20/package.json
@@ -16,7 +16,6 @@
     "@angular/core": "^20.3.16",
     "@angular/forms": "^20.0.0",
     "@angular/platform-browser": "^20.0.0",
-    "@angular/platform-browser-dynamic": "^20.0.0",
     "@angular/router": "^20.0.0",
     "angular-in-memory-web-api": "^0.20.0",
     "rxjs": "~7.8.2",

--- a/package.json
+++ b/package.json
@@ -62,7 +62,6 @@
     "@angular/compiler-cli": ">=20.0.0 <23.0.0",
     "@angular/core": ">=20.0.0 <23.0.0",
     "@angular/platform-browser": ">=20.0.0 <23.0.0",
-    "@angular/platform-browser-dynamic": ">=20.0.0 <23.0.0",
     "jest": "^30.0.0",
     "jsdom": ">=26.0.0",
     "typescript": ">=5.8"
@@ -78,7 +77,6 @@
     "@angular/forms": "^22.0.1",
     "@angular/material": "^22.0.1",
     "@angular/platform-browser": "^22.0.1",
-    "@angular/platform-browser-dynamic": "^22.0.1",
     "@babel/core": "^7.29.7",
     "@babel/preset-env": "^7.29.7",
     "@commitlint/cli": "^21.0.2",

--- a/setup-env/zone/index.js
+++ b/setup-env/zone/index.js
@@ -1,13 +1,9 @@
 require('zone.js');
 require('zone.js/testing');
 
-const { COMPILER_OPTIONS, NgModule, provideZoneChangeDetection, VERSION } = require('@angular/core');
+const { COMPILER_OPTIONS, NgModule, provideZoneChangeDetection } = require('@angular/core');
 const { getTestBed } = require('@angular/core/testing');
 const { BrowserTestingModule, platformBrowserTesting } = require('@angular/platform-browser/testing');
-const {
-    BrowserDynamicTestingModule,
-    platformBrowserDynamicTesting,
-} = require('@angular/platform-browser-dynamic/testing');
 
 const { polyfillEncoder, resolveTestEnvOptions } = require('../utils');
 
@@ -15,44 +11,24 @@ const setupZoneTestEnv = (options) => {
     polyfillEncoder();
     const resolvedOptions = resolveTestEnvOptions(options) ?? {};
     const { extraProviders = [], ...testEnvironmentOptions } = resolvedOptions;
-    if (+VERSION.major >= 21) {
-        class TestModule {}
-        NgModule({
-            providers: [provideZoneChangeDetection()],
-        })(TestModule);
 
-        getTestBed().initTestEnvironment(
-            [BrowserTestingModule, TestModule],
-            platformBrowserTesting([
-                {
-                    provide: COMPILER_OPTIONS,
-                    useValue: {},
-                    multi: true,
-                },
-                ...extraProviders,
-            ]),
-            testEnvironmentOptions,
-        );
-    } else if (+VERSION.major === 20) {
-        getTestBed().initTestEnvironment(
-            [BrowserTestingModule],
-            platformBrowserTesting([
-                {
-                    provide: COMPILER_OPTIONS,
-                    useValue: {},
-                    multi: true,
-                },
-                ...extraProviders,
-            ]),
-            testEnvironmentOptions,
-        );
-    } else {
-        getTestBed().initTestEnvironment(
-            [BrowserDynamicTestingModule],
-            platformBrowserDynamicTesting(extraProviders),
-            testEnvironmentOptions,
-        );
-    }
+    class TestModule {}
+    NgModule({
+        providers: [provideZoneChangeDetection()],
+    })(TestModule);
+
+    getTestBed().initTestEnvironment(
+        [BrowserTestingModule, TestModule],
+        platformBrowserTesting([
+            {
+                provide: COMPILER_OPTIONS,
+                useValue: {},
+                multi: true,
+            },
+            ...extraProviders,
+        ]),
+        testEnvironmentOptions,
+    );
 };
 
 module.exports = {

--- a/setup-env/zone/index.mjs
+++ b/setup-env/zone/index.mjs
@@ -1,10 +1,9 @@
 import 'zone.js';
 import 'zone.js/testing';
 
-import { COMPILER_OPTIONS, NgModule, provideZoneChangeDetection, VERSION } from '@angular/core';
+import { COMPILER_OPTIONS, NgModule, provideZoneChangeDetection } from '@angular/core';
 import { getTestBed } from '@angular/core/testing';
 import { BrowserTestingModule, platformBrowserTesting } from '@angular/platform-browser/testing';
-import { BrowserDynamicTestingModule, platformBrowserDynamicTesting } from '@angular/platform-browser-dynamic/testing';
 
 import { polyfillEncoder, resolveTestEnvOptions } from '../utils';
 
@@ -12,44 +11,24 @@ const setupZoneTestEnv = (options) => {
     polyfillEncoder();
     const resolvedOptions = resolveTestEnvOptions(options) ?? {};
     const { extraProviders = [], ...testEnvironmentOptions } = resolvedOptions;
-    if (+VERSION.major >= 21) {
-        class TestModule {}
-        NgModule({
-            providers: [provideZoneChangeDetection()],
-        })(TestModule);
 
-        getTestBed().initTestEnvironment(
-            [BrowserTestingModule, TestModule],
-            platformBrowserTesting([
-                {
-                    provide: COMPILER_OPTIONS,
-                    useValue: {},
-                    multi: true,
-                },
-                ...extraProviders,
-            ]),
-            testEnvironmentOptions,
-        );
-    } else if (+VERSION.major === 20) {
-        getTestBed().initTestEnvironment(
-            [BrowserTestingModule],
-            platformBrowserTesting([
-                {
-                    provide: COMPILER_OPTIONS,
-                    useValue: {},
-                    multi: true,
-                },
-                ...extraProviders,
-            ]),
-            testEnvironmentOptions,
-        );
-    } else {
-        getTestBed().initTestEnvironment(
-            [BrowserDynamicTestingModule],
-            platformBrowserDynamicTesting(extraProviders),
-            testEnvironmentOptions,
-        );
-    }
+    class TestModule {}
+    NgModule({
+        providers: [provideZoneChangeDetection()],
+    })(TestModule);
+
+    getTestBed().initTestEnvironment(
+        [BrowserTestingModule, TestModule],
+        platformBrowserTesting([
+            {
+                provide: COMPILER_OPTIONS,
+                useValue: {},
+                multi: true,
+            },
+            ...extraProviders,
+        ]),
+        testEnvironmentOptions,
+    );
 };
 
 export { setupZoneTestEnv };

--- a/setup-env/zoneless/index.js
+++ b/setup-env/zoneless/index.js
@@ -2,10 +2,6 @@ const angularCore = require('@angular/core');
 const { ErrorHandler, NgModule, VERSION, COMPILER_OPTIONS } = require('@angular/core');
 const { getTestBed } = require('@angular/core/testing');
 const { BrowserTestingModule, platformBrowserTesting } = require('@angular/platform-browser/testing');
-const {
-    platformBrowserDynamicTesting,
-    BrowserDynamicTestingModule,
-} = require('@angular/platform-browser-dynamic/testing');
 
 const { polyfillEncoder, resolveTestEnvOptions } = require('../utils');
 const provideZonelessChangeDetectionFn =
@@ -37,26 +33,18 @@ const setupZonelessTestEnv = (options) => {
         polyfillEncoder();
         const resolvedOptions = resolveTestEnvOptions(options) ?? {};
         const { extraProviders = [], ...testEnvironmentOptions } = resolvedOptions;
-        if (+VERSION.major >= 20) {
-            getTestBed().initTestEnvironment(
-                [BrowserTestingModule, provideZonelessConfig()],
-                platformBrowserTesting([
-                    {
-                        provide: COMPILER_OPTIONS,
-                        useValue: {},
-                        multi: true,
-                    },
-                    ...extraProviders,
-                ]),
-                testEnvironmentOptions,
-            );
-        } else {
-            getTestBed().initTestEnvironment(
-                [BrowserDynamicTestingModule, provideZonelessConfig()],
-                platformBrowserDynamicTesting(extraProviders),
-                testEnvironmentOptions,
-            );
-        }
+        getTestBed().initTestEnvironment(
+            [BrowserTestingModule, provideZonelessConfig()],
+            platformBrowserTesting([
+                {
+                    provide: COMPILER_OPTIONS,
+                    useValue: {},
+                    multi: true,
+                },
+                ...extraProviders,
+            ]),
+            testEnvironmentOptions,
+        );
 
         return;
     }

--- a/setup-env/zoneless/index.mjs
+++ b/setup-env/zoneless/index.mjs
@@ -2,7 +2,6 @@ import * as angularCore from '@angular/core';
 import { ErrorHandler, NgModule, VERSION, COMPILER_OPTIONS } from '@angular/core';
 import { getTestBed } from '@angular/core/testing';
 import { BrowserTestingModule, platformBrowserTesting } from '@angular/platform-browser/testing';
-import { platformBrowserDynamicTesting, BrowserDynamicTestingModule } from '@angular/platform-browser-dynamic/testing';
 
 import { polyfillEncoder, resolveTestEnvOptions } from '../utils';
 
@@ -36,26 +35,18 @@ const setupZonelessTestEnv = (options) => {
         polyfillEncoder();
         const resolvedOptions = resolveTestEnvOptions(options) ?? {};
         const { extraProviders = [], ...testEnvironmentOptions } = resolvedOptions;
-        if (+VERSION.major >= 20) {
-            getTestBed().initTestEnvironment(
-                [BrowserTestingModule, provideZonelessConfig()],
-                platformBrowserTesting([
-                    {
-                        provide: COMPILER_OPTIONS,
-                        useValue: {},
-                        multi: true,
-                    },
-                    ...extraProviders,
-                ]),
-                testEnvironmentOptions,
-            );
-        } else {
-            getTestBed().initTestEnvironment(
-                [BrowserDynamicTestingModule, provideZonelessConfig()],
-                platformBrowserDynamicTesting(extraProviders),
-                testEnvironmentOptions,
-            );
-        }
+        getTestBed().initTestEnvironment(
+            [BrowserTestingModule, provideZonelessConfig()],
+            platformBrowserTesting([
+                {
+                    provide: COMPILER_OPTIONS,
+                    useValue: {},
+                    multi: true,
+                },
+                ...extraProviders,
+            ]),
+            testEnvironmentOptions,
+        );
 
         return;
     }

--- a/src/config/setup-env.spec.ts
+++ b/src/config/setup-env.spec.ts
@@ -27,28 +27,22 @@ jest.mock('@angular/core/testing', () => {
 });
 
 class BrowserTestingModuleStub {}
-class BrowserDynamicTestingModuleStub {}
 class PlatformRefStub {}
 class ErrorHandlerStub {}
 const mockPlatformBrowserTesting = jest.fn(() => new PlatformRefStub());
-const mockPlatformBrowserDynamicTesting = jest.fn(() => new PlatformRefStub());
 jest.mock('@angular/platform-browser/testing', () => {
     return {
         BrowserTestingModule: new BrowserTestingModuleStub(),
         platformBrowserTesting: mockPlatformBrowserTesting,
     };
 });
-jest.mock('@angular/platform-browser-dynamic/testing', () => {
-    return {
-        BrowserDynamicTestingModule: new BrowserDynamicTestingModuleStub(),
-        platformBrowserDynamicTesting: mockPlatformBrowserDynamicTesting,
-    };
-});
 const mockProvideExperimentalZonelessChangeDetection = jest.fn();
+const mockProvideZoneChangeDetection = jest.fn();
 const mockCompilerOptions = 'COMPILER_OPTIONS';
 jest.mock('@angular/core', () => {
     return {
         provideExperimentalZonelessChangeDetection: mockProvideExperimentalZonelessChangeDetection,
+        provideZoneChangeDetection: mockProvideZoneChangeDetection,
         ErrorHandler: ErrorHandlerStub,
         NgModule: () => {
             return jest.fn();

